### PR TITLE
fix: hasattr checks if attr is None

### PR DESCRIPTION
**Pull request by mxyng**
```
>>> import ollama
>>> r = ollama.chat(model='tmp')
>>> r.message
Message(role='assistant', content='', images=None, tool_calls=None)
>>> 'tool_calls' in r.message
False
>>> 'role' in r.message
True
```

### CODE REVIEW

**Review by jmorganca — APPROVED** [review]

**Review by jmorganca — APPROVED** [review]

File: ollama/_types.py [hunk 1]
```diff
@@ -23,7 +23,30 @@ def __setitem__(self, key: str, value: Any) -> None:
...
+    >>> 'tool_calls' in msg
+    False
+    >>> msg['tool_calls'] = []
+    >>> 'tool_calls' in msg
+    True
+    >>> msg['tool_calls'] = [Message.ToolCall(function=Message.ToolCall.Function(name='foo', arguments={}))]
+    >>> 'tool_calls' in msg
+    True
+    """
+    return key in self.model_fields_set
```
Comment (jmorganca) [inline ollama/_types.py:49]: Will this inadvertently filter out fields with a default value?

Comment (mxyng) [reply ollama/_types.py:49]: I don't think so

> Returns the set of fields that have been explicitly set on this model instance.

Comment (jmorganca) [reply ollama/_types.py:49]: Did we want this?

```
return key in self.model_fields_set or key is not None
```

I don't think default values are in `model_fields_set`

```
@property
    def model_fields_set(self) -> set[str]:
        """Returns the set of fields that have been explicitly set on this model instance.

        Returns:
            A set of strings representing the fields that have been set,
                i.e. that were not filled from defaults.
        """
        return self.__pydantic_fields_set__
```

Comment (ParthSareen) [reply ollama/_types.py:49]: Screenshot 2024-11-23 at 6 03 36 PM
Just checked - default values are omitted in this case which is not what we want.

Comment (jmorganca) [inline ollama/_types.py:49]:
```suggestion
    if key in self.model_fields_set:
        return True

    if key in self.model_fields:
        return self.model_fields[key].default is not None

    return False
```

**Review by ParthSareen — APPROVED** [review]

**Pull request merged**